src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs

```diff
@@ -97,20 +96,28 @@ private void Handle(GetPooledTransactionsMessage msg)
                              $"in {stopwatch.Elapsed.TotalMilliseconds}ms");
         }
 
-        protected PooledTransactionsMessage FulfillPooledTransactionsRequest(
+        internal PooledTransactionsMessage FulfillPooledTransactionsRequest(
             GetPooledTransactionsMessage msg)
         {
-            List<Transaction> txs = new();
-            int responseSize = Math.Min(256, msg.Hashes.Count);
-            for (int i = 0; i < responseSize; i++)
+            int packetSizeLeft = TransactionsMessage.MaxPacketSize;
+            List<Transaction> txsToSend = new();
```

Comment:
Preallocated list size.

We could use `ArrayPoolList` if message would eventually get disposed.

Reply:
That was my first thought, but needed to change it to standard list because of `ObjectDisposedException`. I didn't used `ArrayPoolList` correctly or there is some bug in it.

When it was:
`using ArrayPoolList<Transaction> txsToSend = new(1024);`
instead of:
`List<Transaction> txsToSend = new();`

I had this exception in hive eth protocol tests:
```
Error in communication with [Node|     172.17.0.3:37690|eth68|]: System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'ArrayPoolList'.
   at Nethermind.Core.Collections.ArrayPoolList`1.GuardIndex(Int32 index, Boolean shouldThrow, Boolean allowEqualToCount) in /src/Nethermind/Nethermind.Core/Collections/ArrayPoolList.cs:line 162
   at Nethermind.Core.Collections.ArrayPoolList`1.get_Item(Int32 index) in /src/Nethermind/Nethermind.Core/Collections/ArrayPoolList.cs:line 150
   at Nethermind.Network.P2P.Subprotocols.Eth.V62.Messages.TransactionsMessageSerializer.GetLength(TransactionsMessage message, Int32& contentLength) in /src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V62/Messages/TransactionsMessageSerializer.cs:line 39
   at Nethermind.Network.P2P.Subprotocols.Eth.V65.Messages.PooledTransactionsMessageSerializer.GetLength(PooledTransactionsMessage message, Int32& contentLength) in /src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Messages/PooledTransactionsMessageSerializer.cs:line 28
   at Nethermind.Network.P2P.Subprotocols.Eth.V66.Messages.Eth66MessageSerializer`2.GetLength(TEth66Message message, Int32& contentLength) in /src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V66/Messages/Eth66MessageSerializer.cs:line 43
   at Nethermind.Network.MessageSerializationService.ZeroSerialize[T](T message)
   at Nethermind.Network.P2P.PacketSender.Enqueue[T](T message) in /src/Nethermind/Nethermind.Network/P2P/PacketSender.cs:line 41
   at Nethermind.Network.P2P.Session.DeliverMessage[T](T message) in /src/Nethermind/Nethermind.Network/P2P/Session.cs:line 222
   at Nethermind.Network.P2P.ProtocolHandlers.ProtocolHandlerBase.Send[T](T message) in /src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/ProtocolHandlerBase.cs:line 82
   at Nethermind.Network.P2P.Subprotocols.Eth.V66.Eth66ProtocolHandler.Handle(GetPooledTransactionsMessage getPooledTransactions) in /src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V66/Eth66ProtocolHandler.cs:line 152
   at Nethermind.Network.P2P.Session.ReceiveMessage(ZeroPacket zeroPacket) in /src/Nethermind/Nethermind.Network/P2P/Session.cs:line 200
   at Nethermind.Network.P2P.ProtocolHandlers.ZeroNettyP2PHandler.ChannelRead0(IChannelHandlerContext ctx, ZeroPacket input) in /src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/ZeroNettyP2PHandler.cs:line 99
   at DotNetty.Transport.Channels.SimpleChannelInboundHandler`1.ChannelRead(IChannelHandlerContext ctx, Object msg)
   at DotNetty.Transport.Channels.AbstractChannelHandlerContext.InvokeChannelRead(Object msg) 
```

I dropped `ArrayPoolList` in the latest commit

Reply:
You would have to initialize `ArrayPoolList` and Dispose it in the same method you use `Send` so method above this one.